README.md

```diff
@@ -1,2 +1,288 @@
-# Balance On Wheels
-Modeling and Control of Two-Legged Wheeled Robot: In this project, we will be building a two wheeled robot (with a hip and knee joint) which can balance at variable heights including a provision for jumping.
+![](https://i.imgur.com/V6Vat3e.jpg)
+
+# <ins>Table of Contents
+
+* [About the Project](#about-the-project)
+  * [Tech Stack](#tech-stack)
+  * [File Structure](#file-structure)
+* [Getting Started](#getting-started)
+  * [Prerequisites](#prerequisites)
+  * [Installations](#installations)
+  * [Execution](#execution)
+* [Theory and Approach](#theory-and-approach)
+* [Results and Demo](#results-and-demo)
+* [Future Work](#future-work)
+* [Troubleshooting](#troubleshooting)
+* [Contributors](#contributors)
+* [Acknowledgements and Resources](#acknowledgements-and-resources)
+* [License](#license)
+
+---
+
+# <ins>About the Project
+
+**Modeling and Control of Two-Legged Wheeled Robot: In this project, we will be building a two wheeled robot (with a hip and knee joint) which can balance at variable heights including a provision for jumping.**
+
+---
+
+## <ins>Tech Stack
+
+* <font color="red">C</font>
+* <font color="red">Python</font>
+* <font color="red">SolidWorks</font>
+* <font color="red">ESP-IDF</font>
+* <font color="red">AutoCad</font>
+* <font color="red">Creality</font>
+
+## <ins>File Structure
+```
+- myProject/
+             - CMakeLists.txt
+             - sdkconfig
+             - components/ - component1/ - CMakeLists.txt
+                                         - Kconfig
+                                         - src1.c
+                           - component2/ - CMakeLists.txt
+                                         - Kconfig
+                                         - src1.c
+                                         - include/ - component2.h
+             - main/       - CMakeLists.txt
+                           - src1.c
+                           - src2.c
+
+             - build/
+```
+---
+# <ins>Getting Started
+
+## <ins>Prerequisites
```

Comment:
Mention versions of software used like ESP_IDF v4.4 etc

Reply:
made the required changes